code/crates/engine/src/consensus.rs

```diff
@@ -1248,7 +1220,36 @@ where
                 Ok(r.resume_with(()))
             }
 
-            Effect::SyncValue(value, r) => {
+            Effect::InvalidValue(peer, height, error, r) => {
+                if let Some(sync) = &self.sync {
+                    if let ConsensusError::InvalidCommitCertificate(certificate, e) = error {
+                        error!(
+                            %peer,
+                            %certificate.height,
+                            %certificate.round,
+                            "Invalid certificate received: {e}"
+                        );
+
+                        sync.cast(SyncMsg::InvalidValue(peer, certificate.height))
+                            .map_err(|e| {
+                                eyre!("Error when notifying sync of invalid certificate: {e}")
+                            })?;
+                    } else {
+                        sync.cast(SyncMsg::ValueProcessingError(peer, height))
+                            .map_err(|e| {
+                                eyre!("Error when notifying sync of value processing error: {e}")
+                            })?;
+                    }
+                }
+
+                Ok(r.resume_with(()))
+            }
+
+            Effect::SyncValue(value, proposer, r) => {
+                // NOTE: The state.height is not yet updated if this is an effect that is triggered by the
+                // Msg::StartHeight(height), with buffered sync value for height `height`. So this will panic.
+                // assert_eq!(value.certificate.height, state.height);
```

Comment:
Do you find it useful to leave the commented out assertion here or should we remove it?

Reply:
no, i meant to remove after discussing with you. It was pretty difficult to debug the cause of assert failure, not sure if there is anything we can do about it (have the `state.height` updated somehow), couldn't figure out.